[Turnstile] Q4 updates, feature table, ephemeral IDs (#18721)
cloudflare/cloudflare-docs: src/content/docs/turnstile/concepts/hostname-management.mdx

---
title: Hostname management
pcx_content_type: reference
sidebar:
  order: 2

---

By default, all widgets can have up to 10 hostnames associated with a widget. A widget requires at least one hostname to be entered.

> Only Enterprise Bot Management and Enterprise Turnstile customers can have this limit increased. Contact your account team to increase your hostname limit.

Reviewer: ```suggestion
Customers with Enterprise Bot Management or Enterprise Turnstile can have this limit increased. Contact your account team to increase your hostname limit.
```
Improved wording imo

a few more thoughts:
- Turnstile is separate from BM now?
- maybe link to sales?
- Also add to FAQ?

You must specify a list of hostnames when creating a widget. The widget can only be used on these hostnames and will not work on any other hostnames. You can use subdomains to restrict the widgets further.

The hostname should not contain a scheme `http://` or `https://`, a port `443`, or a path `/`.

Specifying a subdomain is optional.

For example, using the `www.example.com` value will allow widgets on the following hostnames:

- `www.example.com`
- `abc.www.example.com:8080`

but not on the following hostnames:

- `example.com`
- `dash.example.com`
- `cloudflare.com`

When the widget is embedded on a hostname not listed, it will show an error message.

:::note

> Enterprise Bot Management customers can remove the hostname validation requirement for a widget. For more information on this feature, contact your account team. 

Reviewer: What about Enterprise Turnstile here? Also this is duplicated by the section right below, so either remove this aside or the next section
